Fix typos, headers, links in Semgrep Pro Engine examples doc (#888)
returntocorp/semgrep-docs: docs/deepsemgrep/deepsemgrep-examples.md

<iframe title="Semgrep example no prints" src="https://semgrep.dev/embed/editor?snippet=Po9p" width="100%" height="432" frameborder="0"></iframe>

Semgrep Pro matches both dangerous calls because it does cross function boundaries. In fact, with Semgrep Pro, the taint rule can track calls to `get_user_input` over multiple jumps in multiple files.

:::tip Try it out
Enable the **Semgrep Pro Engine beta** <i class="fa-solid fa-toggle-large-on"></i> toggle in the following link to an [example of dangerous taint](https://semgrep.dev/s/Po9p). To run Semgrep Pro Engine in the cloned [Semgrep Pro Engine testing repository](https://github.com/returntocorp/semgrep-pro-tests). Go to `docs/taint_tracking/javascript` and run the following command:

```sh
semgrep --config pro.yaml . --pro
```
:::

#### ES6 and CommonJS

The JavaScript and TypeScript ecosystems contain various ways for importing and exporting code, Semgrep Pro Engine can track dataflow through ES6 imports or exports and some CommonJS export paths (See known limitations below).

##### ES6

> TODO Semgrep is able to track data through the definition of exports for es6:

Author: @milanwilliams please, instead of just Semgrep, is this Semgrep OSS or Semgrep Pro Engine? (We need to be specific.)

Reviewer: Semgrep Pro Engine 👍

Author: ```suggestion
Semgrep Pro Engine is able to track data through the definition of exports for es6:
```

> Semgrep Pro Engine is able to track data through the definition of exports for es6:

Author: ```suggestion
Semgrep Pro Engine can track data through the definition of exports for es6:
```

```js
export function readUser() {
    return get_user_input("example")
}
```

> Semgrep is able to follow the dataflow when it is imported into another location:

Author: @milanwilliams please, instead of just Semgrep, is this Semgrep OSS or Semgrep Pro Engine? (We need to be specific.)

Author: ```suggestion
Semgrep Pro Engine is able to follow the dataflow when it is imported into another location:
```

> Semgrep Pro Engine is able to follow the dataflow when it is imported into another location:

Author: ```suggestion
Semgrep Pro Engine can follow the dataflow when it is imported into another location:
```

```js
import { readUser } from "./es6/es6";

readUser()
```

##### CommonJS

> TODO Semgrep is able to track data through the definition of exports for CommonJS when the function is defined inline:

Author: @milanwilliams please, instead of just Semgrep, is this Semgrep OSS or Semgrep Pro Engine? (We need to be specific.)

Reviewer: Semgrep Pro Engine 👍

Author: ```suggestion
Semgrep Pro Engine is able to track data through the definition of exports for CommonJS when the function is defined inline:
```

> Semgrep Pro Engine is able to track data through the definition of exports for CommonJS when the function is defined inline:

Author: ```suggestion
Semgrep Pro Engine can track data through the definition of exports for CommonJS when the function is defined inline:
```

#### ES6 and CommonJS

> The JavaScript and TypeScript ecosystems contain various ways for importing and exporting code, Semgrep Pro Engine is can track dataflow through ES6 imports or exports and some CommonJS export paths (See known limitations below).

Author: ```suggestion
The JavaScript and TypeScript ecosystems contain various ways for importing and exporting code, Semgrep Pro Engine can track dataflow through ES6 imports or exports and some CommonJS export paths (See known limitations below).
```

semgrep --config pro.yaml . --pro
```

> #### JavaScript an TypeScript

Reviewer 2: and

Author: Thank you!

```suggestion
#### JavaScript and TypeScript
```
